### PR TITLE
Fix issue with temp path on different file system

### PR DESCRIPTION
**Pull request by davidiw**
This addresses the case where our temp file system isn't on the same file system as the location where safety rules is. The solution is to generate the temp file in the same path as safety rules.

About to start a cluster test to verify it works. It works.

### CODE REVIEW

File: common/temppath/src/lib.rs [hunk 1]
```diff
@@ -28,11 +28,16 @@ impl Drop for TempPath {
...
+    /// Create new uninitialized temporary path in the system temp directory, note: a file /
+    /// directory isn't created automatically
     pub fn new() -> Self {
-        // Create a random path in the system temp directory
-        let mut temppath = std::env::temp_dir();
+        Self::new_with_temp_dir(std::env::temp_dir())
+    }
+
+    /// Create new uninitialized temporary path in the specified directory, note: a file /
+    /// directory isn't created automatically
```
Comment (aching) [inline common/temppath/src/lib.rs]: Perhaps, this statement will be less ambiguous.

/// Create new uninitialized temporary path in the specified directory, note that 
/// the temp_dir directory isn't created automatically

File: secure/storage/src/on_disk.rs [hunk 2]
```diff
@@ -28,9 +28,14 @@ impl OnDiskStorage {
             File::create(&file_path).expect("Unable to create storage");
         }
 
+        // If there is no path just a file use the same.
+        let file_dir = file_path
+            .parent()
+            .map_or(PathBuf::new(), |p| p.to_path_buf());
+
```
Comment (aching) [inline secure/storage/src/on_disk.rs]: Curious why we should support this behavior?  Seems like a special case we shouldn't expect?

Comment (davidiw) [reply secure/storage/src/on_disk.rs]: Improved comment.

**Pull request closed without merging**